src/main/java/de/ait/secondlife/controllers/CategoryController.java

```diff
@@ -0,0 +1,54 @@
+package de.ait.secondlife.controllers;
+
+import de.ait.secondlife.dto.CategoryDto;
+import de.ait.secondlife.dto.IsActiveCategoryDto;
+import de.ait.secondlife.services.interfaces.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/v1/categories")
+@Tag(name = "Category controller")
+public class CategoryController {
+
+    private final CategoryService service;
+
+    @Operation(summary = "Get category by id")
+    @GetMapping("/{category-id}")
+    public ResponseEntity<CategoryDto> getById(@PathVariable("category-id")Long categoryId){
+        return ResponseEntity.ok(service.getById(categoryId));
+    }
+
+    @Operation(summary = "Get list of the categories")
+    @GetMapping
+    public ResponseEntity<List<CategoryDto>> getAll(){
+        return ResponseEntity.ok(service.getAll());
+    }
+
+
+    @Operation(summary = "Add category")
+    @PostMapping
+    public ResponseEntity<CategoryDto> add(@RequestBody CategoryDto dto){
```

Comment:
there should be a separate Dto (NewCategoryDto for example) or we should ignore id parameter from the request. 
the current implementation of endpoint allows not only create but also update of existing records

Reply:
Done